test/Tests/ApiSpec.cs

```diff
@@ -94,44 +67,134 @@ public async Task Fail_to_create_auction(string sample, int index)
         Assert.Equal(HttpStatusCode.NotFound, bidResponse.StatusCode);
     }
+}
+public class ApiAsyncSpecJwtToken:ApiAsyncSpec<JwtApiAuth>{}
+public class ApiAsyncSpecMsClientPrincipal:ApiAsyncSpec<MsClientPrincipalApiAuth>{}
+public abstract class ApiAsyncSpec<TAuth>
+    where TAuth:IApiAuth, new()
+{
+    [Fact]
+    public async Task Create_auction_1()
+    {
+        using var application = new AsyncApiFixture<TAuth>(GetType().Name+"_"+nameof(Create_auction_1)+".db");
+        var response = await application.PostAuction(FirstAuctionRequest, AuthToken.Seller1);
+        var auctionResponse = await application.GetAuction(1, AuthToken.Seller1);
+        var stringContent = await auctionResponse.Content.ReadAsStringAsync();
+        Assert.Multiple(() =>
+        {
+            Assert.Equal(HttpStatusCode.Accepted,response.StatusCode);
+            Assert.Equal(HttpStatusCode.OK, auctionResponse.StatusCode);
+            Assert.Equal(JToken.Parse(FirstAuctionResponse).ToString(Formatting.Indented),
+                JToken.Parse(stringContent).ToString(Formatting.Indented));
+        });
+    }
+
+    [Fact]
+    public async Task Create_auction_2()
+    {
+        using var application = new AsyncApiFixture<TAuth>(GetType().Name+"_"+nameof(Create_auction_2)+".db");
+        var response = await application.PostAuction(SecondAuctionRequest, AuthToken.Seller1);
+        var auctionResponse = await application.GetAuction(1, AuthToken.Seller1);
+        var stringContent = await auctionResponse.Content.ReadAsStringAsync();
+        Assert.Multiple(() =>
+        {
+            Assert.Equal(HttpStatusCode.Accepted,response.StatusCode);
+            Assert.Equal(HttpStatusCode.OK, auctionResponse.StatusCode);
+            Assert.Equal(JToken.Parse(SecondAuctionResponse).ToString(Formatting.Indented),
+                JToken.Parse(stringContent).ToString(Formatting.Indented));
+        });
+    }
+    [Theory,
+     InlineData(@"{
+        ""startsAt"": ""2021-12-01T10:00:00.000Z"",
+        ""endsAt"": ""2022-12-18T10:00:00.000Z"",
+        ""title"": ""Some auction"",
+        ""currency"": ""VAC"",
+        ""timeFrame"": -1,
+        ""reservePrice"": null,
+        ""minRaise"":null
+    }",1),
+     InlineData(@"{
+        ""startsAt"": ""2021-12-01T10:00:00.000Z"",
+        ""endsAt"": ""2022-12-18T10:00:00.000Z"",
+        ""title"": ""Some auction"",
+    }",2),
+    ]
+    public async Task Fail_to_create_auction(string sample, int index)
+    {
+        using var application = new AsyncApiFixture<TAuth>(GetType().Name+"_"+nameof(Fail_to_create_auction)+"_"+index+".db");
+        var response = await application.PostAuction(sample, AuthToken.Seller1);
+        Assert.Multiple(() =>
+        {
+            Assert.Equal(HttpStatusCode.BadRequest,response.StatusCode);
+        });
+    }
+    [Fact]
+    public async Task Place_bid_as_buyer_on_auction_1()
```

Comment:
Almost the same flow be used to test sync version as well, only difference is OK instead of Accepted